### DISCUSSION TITLE
Automatic git pull in a Codespace

### DISCUSSION THREAD

**Discussion by alexavil** [community/community#17324] [Codespaces]
Similarly to how a Codespace does `npm install` on start, it should do `git pull` either:
- After every X minutes (could be changed by the user)
- On Codespace restart
- On both of the above
The user should be able to toggle this feature on and off via Command Pallete or `.devcontainer.json`. This could be useful if you mainly use Codespaces for testing or have to code outside of a Codespace for some other reason.

**Comment by legomushroom** [comment]
Hey @alexavil 👋 Thanks for the feedback! 

You can turn on the periodic fetch in vscode settings, would it suffice in your case?

- open command palette (`shift + meta + p`) and search for `Git: Sync` command
- run the command, the `Would you like Code to periodically run 'git fetch'`, click `Yes` button:

<img width="1904" alt="image" src="https://user-images.githubusercontent.com/1478800/170566898-52eb9b3f-85bb-4bcb-9947-93ddd4b37811.png">

You can fine-tune git sync behaviour in the settings of the `Git` extension (search for `Git: sync` in the settings):

<img width="1904" alt="image" src="https://user-images.githubusercontent.com/1478800/170567209-35b1d343-4339-48d4-a6f1-c01c96476df7.png">

You can also turn on settings sync on (click on `cog` icon in bottom left corner -> `Settings sync is off`) to enable it by default for all clients:

<img width="1904" alt="image" src="https://user-images.githubusercontent.com/1478800/170569123-f6db65e5-6467-45b6-bb6f-5c498ac4c8f8.png">

It might be hard to enable this for all users by default, because some folks use huge monorepos with codespaces hence automatic git sync would be very invasive for their workflow 🤔

**Reply by alexavil** [reply]
IIRC `git fetch` only gets data about branches and commits, but doesn't merge them.
If you're unhappy with `git pull` being default, it could be done the same way as `git fetch`.

**Reply by alexavil** [reply]
Correct me if I'm wrong.
